SIT.Manager/Services/AkiServerRequestingService.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        AkiServer ret = new(serverAddresss);
+        if (fetchInformation)
+        {
+            AkiServerInfo? serverInfo = await GetAkiServerInfoAsync(ret, cancellationToken);
+            if(serverInfo != null)
+            {
+                ret.Name = serverInfo.Name;
+            }
+        }
+
+        return ret;
+    }
+
+    public async Task<int> GetPingAsync(AkiServer akiServer, CancellationToken cancellationToken = default)
+    {
+        var strategy = _resiliencePipelineProvider.GetPipeline<HttpResponseMessage>("ping-pipeline");
+        Stopwatch stopwatch = Stopwatch.StartNew();
+        using MemoryStream respStream = await SendAsync(akiServer, "/launcher/ping", strategy: strategy, cancellationToken: cancellationToken);
+        stopwatch.Stop();
+
+        using (StreamReader streamReader = new(respStream))
+        {
+            if (streamReader.ReadToEnd().Equals("\"pong!\"", StringComparison.InvariantCultureIgnoreCase))
+                return Convert.ToInt32(stopwatch.ElapsedMilliseconds);
+            else
+                return -1;
+        }
+    }
+
+    public async Task<List<AkiMiniProfile>> GetMiniProfilesAsync(AkiServer server, CancellationToken cancellationToken = default)
+    {
+        using MemoryStream respStream = await SendAsync(server, "/launcher/profiles", cancellationToken: cancellationToken);
+        using StreamReader streamReader = new(respStream);
+        return JsonConvert.DeserializeObject<List<AkiMiniProfile>>(await streamReader.ReadToEndAsync(cancellationToken)) ?? [];
+    }
+
+    private string CreateLoginData(AkiCharacter character)
+    {
+        Version SITVersion = new(_configService.Config.SitVersion);
+        string compatibleUri = character.ParentServer.Address.AbsoluteUri[..^(SITVersion >= standardUriFormatSupportedVersion ? 0 : 1)];
+        JsonObject loginData = new()
+        {
+            ["username"] = character.Username,
+            ["email"] = character.Username,
+            ["edition"] = character.Edition,
+            ["password"] = character.Password,
+            ["backendUrl"] = compatibleUri
+        };
+        return loginData.ToJsonString();
+    }
+
+    private async Task<string> LoginOrRegisterAsync(AkiCharacter character, string operation, CancellationToken cancellationToken = default)
+    {
+        using (MemoryStream ms = await SendAsync(character.ParentServer, Path.Combine("/launcher/profile", operation), HttpMethod.Post, CreateLoginData(character), cancellationToken: cancellationToken))
+        using (StreamReader streamReader = new StreamReader(ms))
+        {
+            string response = await streamReader.ReadToEndAsync(cancellationToken);
+            return response.ToLowerInvariant() switch
+            {
+                "invalid_password" => throw new IncorrectAccountPasswordException(), //Should these really be exceptions?...
+                "failed" => throw new AccountNotFoundException(),
```

Comment:
As your comment suggests they don't necessarily need to be exceptions you could probably achieve the same result by creating an enum with possible outputs and then return an object which contains this enum result and any other info that needs returning then rather than having to handle an exception you just parse the result data.

Which method is better I don't know.